[ONNX] Update the script for version updates

justinchuby opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #83283

This PR updates the `tools/onnx/update_default_opset_version.py` script to ensure files are edited correctly to prepare for the opset 17 support in torch.onnx.

- (clean up) Move script to `main()`
- Add an `--skip_build` option to avoid building pytorch if we want to rerun the process due to errors after compilation is done
- Update to edit the correct files now that the onnx files were refactored

justinchuby committed:
[ONNX] Update the script for version updates

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update the script for version updates"


- Move script to `main()`
- Add an `--skip_recompile` option
- Remove file edits in `__init__` that do not work anymore

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update the script for version updates"


- Move script to `main()`
- Add an `--skip_recompile` option
- Remove file edits in `__init__` that do not work anymore

[ghstack-poisoned]

justinchuby committed:
Update on "[ONNX] Update the script for version updates"


- Move script to `main()`
- Add an `--skip_recompile` option
- Remove file edits in `__init__` that do not work anymore

[ghstack-poisoned]

thiagocrepaldi reviewed (commented):
Hi @justinchuby, please provide a clear and concise description of the bug being handled or the feature/refactored being proposed. The PR description gives context for a faster review. 

If this is a bug fix, describe the bug, provide a minimal repro, describe the strategy to fix and provide unit tests.
If this is a feature/redesign (seems to be the case), outline the motivation for the proposal. Is your code related to a specific problem? e.g., "I'm working on X and would like Y to be possible". If this is related to another GitHub issue, please link here too.

justinchuby commented:
Done. PTAL

thiagocrepaldi commented on tools/onnx/update_default_opset_version.py:
```diff
@@ -77,18 +29,80 @@ def read_sub_write(path: str, prefix_pat: str) -> None:
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--skip_recompile", action="store_true", help="recompile pytorch"
```
```suggestion
        "--skip_build", action="store_true", help="Skips pytorch (re-)build"
```
`recompile` is an unusual term whereas setuptools used by pytorch has targets such as `build`, `install`, `develop`, `bdist` and `clean`. Maybe we should use `build` instead

Also the flag was `--skip_recompile` but the help `recompile pytorch`, one is the opposite of the other :)

justinchuby replied:
Thanks for catching!

justinchuby replied:
PTAL

justinchuby committed:
Update on "[ONNX] Update the script for version updates"


This PR updates the `tools/onnx/update_default_opset_version.py` script to ensure files are edited correctly to prepare for the opset 17 support in torch.onnx.

- (clean up) Move script to `main()`
- Add an `--skip_recompile` option to avoid recompiling pytorch if we want to rerun the process due to errors after compilation is done
- Remove file edits in `__init__` that do not work anymore

[ghstack-poisoned]

AllenTiTaiWang commented on tools/onnx/update_default_opset_version.py:
```diff
@@ -77,18 +29,84 @@ def read_sub_write(path: str, prefix_pat: str) -> None:
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--skip_build", action="store_true", help="Skip building pytorch"
```
Please remember to update the change to your description as well.

justinchuby replied:
Done

titaiwangms reviewed (approved):
🍺

justinchuby committed:
Update on "[ONNX] Update the script for version updates"


This PR updates the `tools/onnx/update_default_opset_version.py` script to ensure files are edited correctly to prepare for the opset 17 support in torch.onnx.

- (clean up) Move script to `main()`
- Add an `--skip_build` option to avoid building pytorch if we want to rerun the process due to errors after compilation is done
- Update to edit the correct files now that the onnx files were refactored

[ghstack-poisoned]

justinchuby commented:
cc @abock for approval

justinchuby commented:
@pytorchbot merge

justinchuby commented:
@pytorchbot rebase

justinchuby committed:
Update on "[ONNX] Update the script for version updates"


This PR updates the `tools/onnx/update_default_opset_version.py` script to ensure files are edited correctly to prepare for the opset 17 support in torch.onnx.

- (clean up) Move script to `main()`
- Add an `--skip_build` option to avoid building pytorch if we want to rerun the process due to errors after compilation is done
- Update to edit the correct files now that the onnx files were refactored

[ghstack-poisoned]

justinchuby commented:
@pytorchbot merge -g
